### PR TITLE
WIP: Prep multichain for experimental release

### PR DESCRIPTION
**Pull request by pimveldhuisen**

### CODE REVIEW

**Comment by Captain-Coder** [comment]
tunnel community tests "test_anon_{tunnel,donwload}" failed for no obvous reason, retest this please.

**Comment by whirm** [comment]
@Captain-Coder you need to start the msg with "retest this please"

**Comment by whirm** [comment]
retest this please

**Comment by Captain-Coder** [comment]
It works when I put it anywhere in the message :) It consistently fails on two tunnel tests. But I'm not sure yet why that is. The test seems to create it's own proxy nodes, so there should be enough of them to build circuits. The changed master member could be a problem if the test needs real world proxies or exits.

**Comment by triblerteam** [comment]
retest this please

**Comment by Captain-Coder** [comment]
Turns out the tests where mixing the HiddenTunnelCommunity with the HiddenTunnelMultichainCommunity. Specifically the main session would have the MC one, and started proxies and exits would use the non MC ones. Thus unable to build circuits. Fixed on PR #2456.

**Comment by synctext** [comment]
nice find!

File: Tribler/community/multichain/database.py [hunk 1]
```diff
@@ -288,7 +288,11 @@ def check_database(self, database_version):
         assert int(database_version) >= 0
         database_version = int(database_version)
 
-        if database_version < 1:
+        if database_version != LATEST_DB_VERSION:
```
Comment (lfdversluis) [inline Tribler/community/multichain/database.py]: I think it's better to check for `database_version < 2` then upgrade to 2. This is in line with all other database upgrade procedures.

Comment (Captain-Coder) [inline Tribler/community/multichain/database.py]: Nice catch, will include this in the new PR. I remember Pim saying something about the version number working oddly for the MC database. (Something to do with it being 0 on initialized environments sometimes?) The only purpose the != serves is to screw things up when someone uses a new database schema with old code, but we shouldn't expect that to work anyway ^.^

**Comment by pimveldhuisen** [comment]
If I understand correctly, https://github.com/Tribler/tribler/pull/2456 makes this PR obsolete?

**Comment by pimveldhuisen** [comment]
This can be closed since it was superseded by  #2456

**Pull request closed without merging**